test/bin/build_images.sh

```diff
@@ -381,7 +382,31 @@ do_group() {
         done
     fi
 
-    if (( ${#buildid_list[@]} )); then
+    if ${BUILD_INSTALLER} && ! ${COMPOSER_DRY_RUN}; then
+        for download_file in "${groupdir}"/*.image-fetcher; do
+            local download_url
+            download_url=$("${GOMPLATE}" --file "${download_file}")
+            blueprint="$(basename -s .image-fetcher "${download_file}")"
+            local expected_iso_file="${VM_DISK_BASEDIR}/${blueprint}.iso"
+            if [ -f "${expected_iso_file}" ]; then
+                echo "${expected_iso_file} already exists"
+                if should_skip "${blueprint}"; then
+                    record_junit "${groupdir}" "${download_file}" "download" "SKIPPED"
+                    continue
+                fi
+            fi
+
+            echo "Adding image-fetcher for ${download_file}"
+            download_opts+=("${expected_iso_file} ${download_url}")
```

Comment:
Downloading the file directly to its final location means that cancelling the download leaves a file in place that prevents the operation from being repeated until the user cleans up that file. It would be more resilient to download to a temporary name/location and then rename the file after it is fully downloaded. I don't know if that's something wget or curl will do automatically or if we would have to wrap the command.